releasenotes/notes/improve-paulievo-power-control-ff73d1ab0d213493.yaml

```diff
@@ -0,0 +1,7 @@
+---
+features_circuits:
+  - |
+    Improved :meth:`.PauliEvolutionGate.control` and :meth:`.PauliEvolutionGate.power` to return
+    more efficient representations in terms of a :class:`.PauliEvolutionGate`. This change leads
+    to significantly shallower circuits and gate counts compared to the previously used
+    generic power and control mechanisms.
```

Comment:
LGTM! 
I only wonder if it makes sense to have some tests on the CX counts, perhaps some ASV benchmarks? or as part of benchpress?